### DISCUSSION TITLE
Accessing Phidget Servo via Hub5000 on MacOS 12.2.1

### DISCUSSION THREAD

**Discussion by srosemer** [artisan-roaster-scope/artisan#868] [Q&A] [answered]
Hello,

I'm looking to integrate a servo motor to control a gas valve but I'm having trouble.  I'm running Artisan 2.6.0 (7088095), did a factory reset, then attempted to interface with a Phidget servo motor, connected to a 16-port servo controller (RCC1000_0) on channel zero, connected to a Phidget wireless hub (HUB5000_0) on port 0.  The wireless hub is connected to my network and I'm able to interface with it, the servo controller, and the servo motor using the Phidget Control Panel, and I've installed all of the drivers available for download.  The host computer is a 2021 MacBook Air, running MacOS 12.2.1.

For the configuration, I configured the Phidget tab in devices for the IP address of the Phidget hub, the password, and the port number.  I tried selecting and unselecting both the "network" checkbox and "remote only" checkboxes, neither which make any difference.  I've also tried using the host name in lieu of the ipaddress, but none of these changes seems to make a difference.

Then in the events tab, I've configured a slider with an RC command with the following:  "set(0, {}, 727171:0); engaged(0, 1, 727171:0)" - with 727171 being the serial number the hub and 0 being the port number that the RC controller is connected to.  The motor is connected to channel zero (hence "0" in the set and engaged commands).  Min/Max values for the slider are 0 and 180 respectively (the range of the servo motor attached).  I've tried the commands both with and without the optional serial number/port - theoretically it should work without that given that there's just the one device, one controller, and one hub.

On the Mac itself, I installed the latest versions of Phidget22 and Phidget22Extra frameworks (I verified their installation in \system\libraries\Frameworks.  I also see the kext file for Phidget installed in the \system\libraries\extensions directory (not sure if that makes a difference on the latest versions of MacOS).  

I don't have the Phidget Control Panel running at the same time as Artisan, I've rebooted my computer as well as the Phidget devices, and I reset Artisan back to factory settings in hopes that these things would change matters, but I'm completely stumped at this point.  Interestingly, I installed Wireshark to watch the network traffic and discovered that there's no network communication between my computer running Artisan and the Phidget device IP address unless I use the Phidget Control Panel.  

Any ideas on what I may be missing or doing incorrectly?

**Comment by MAKOMO** [comment, marked as answer]
> Hello,
> 
> I'm looking to integrate a servo motor to control a gas valve but I'm having trouble. I'm running Artisan 2.6.0 ([7088095](https://github.com/artisan-roaster-scope/artisan/commit/7088095c25d57912eccf502c7c0ffecde682ac58)), did a factory reset, then attempted to interface with a Phidget servo motor, connected to a 16-port servo controller (RCC1000_0) on channel zero, connected to a Phidget wireless hub (HUB5000_0) on port 0. The wireless hub is connected to my network and I'm able to interface with it, the servo controller, and the servo motor using the Phidget Control Panel, and I've installed all of the drivers available for download. The host computer is a 2021 MacBook Air, running MacOS 12.2.1.
> 
> For the configuration, I configured the Phidget tab in devices for the IP address of the Phidget hub, the password, and the port number. I tried selecting and unselecting both the "network" checkbox and "remote only" checkboxes, neither which make any difference. I've also tried using the host name in lieu of the ipaddress, but none of these changes seems to make a difference.
> 

Same setup works for me flawless. I assume Artisan cannot talk to your HUB5000. I would first try this without a password on the HUB5000. The network checkbox needs to be ticked to access Phidgets not physically connected. Keep the HUB5000 on its default port 5661 and specify this also in Artisan. Keep the password field empty in Artisan and type either the IP address as reported by the HUB5000 Web service page or the HUB5000 IP name as reported by the Phidgets control panel (defaults to hub5000.local) under Host.

<img width="673" alt="Screenshot 2022-04-12 at 11 05 31" src="https://user-images.githubusercontent.com/13909011/162930897-719d8c9b-7a29-48a6-bf0c-6ed78cf4554f.png">
<img width="712" alt="Screenshot 2022-04-12 at 11 14 19" src="https://user-images.githubusercontent.com/13909011/162930923-ac325fff-cab7-4e95-a6b5-b9e22403a225.png">
<img width="1056" alt="Screenshot 2022-04-12 at 11 05 11" src="https://user-images.githubusercontent.com/13909011/162930945-9ab63fd5-d205-4666-9983-7742e5bf436f.png">

> Then in the events tab, I've configured a slider with an RC command with the following: "set(0, {}, 727171:0); engaged(0, 1, 727171:0)" - with 727171 being the serial number the hub and 0 being the port number that the RC controller is connected to. The motor is connected to channel zero (hence "0" in the set and engaged commands). Min/Max values for the slider are 0 and 180 respectively (the range of the servo motor attached). I've tried the commands both with and without the optional serial number/port - theoretically it should work without that given that there's just the one device, one controller, and one hub.

As you said, this works without specifying the hub port and serial number if you have only one such device attached.

> 
> On the Mac itself, I installed the latest versions of Phidget22 and Phidget22Extra frameworks (I verified their installation in \system\libraries\Frameworks. I also see the kext file for Phidget installed in the \system\libraries\extensions directory (not sure if that makes a difference on the latest versions of MacOS).
> 
> I don't have the Phidget Control Panel running at the same time as Artisan,

This doesn't matter for remote access.

> I've rebooted my computer as well as the Phidget devices, and I reset Artisan back to factory settings in hopes that these things would change matters, but I'm completely stumped at this point. Interestingly, I installed Wireshark to watch the network traffic and discovered that there's no network communication between my computer running Artisan and the Phidget device IP address unless I use the Phidget Control Panel.
> 
> Any ideas on what I may be missing or doing incorrectly?

Not really. Did you update the firmware on both, the HUB5000 and the RCC1000 to the lastest version using the Phidgets Control Panel? 

I can only assume that Artisan does not successfully attach to your HUB5000. Check the Artisan device log (click the plus icon in the upper link corner and the Artisan log files should get attached to an email). You should see something like this with the corresponding attach message of the RCC1000 and the HUB5000 (here I used my HUB0001):

```
2022-04-12T11:44:48 <Debug> phidget22 : 
	******************************  Logging Enabled  ******************************
	* Phidget22 - Version 1.10 - Built Feb 11 2022 14:05:14                       *
	* Release 1.10 - macOS (64-bit)                                               *
	*******************************************************************************
2022-04-12T11:44:48  <Info> phidget22usb[RawDeviceAdded()] : USB Phidget Attach: 0x00016503 (Vid: 0x06c2 Pid: 0x0043 Version: 111)
2022-04-12T11:44:48  <Info> phidget22[deviceAttach()] : (Attach) HUB0001 (6-Port USB VINT Hub Phidget) v111 S/N:374099
2022-04-12T11:44:48  <Info> phidget22[deviceAttach()] : (Attach) RCC1000 (16x RC Servo Phidget) v111 -> HUB0001 Port:0 S/N:374099
```

**Reply by srosemer** [reply]
Clearing the password field did the trick and thank you for the recommendation to be sure and update the firmware.  Thank you so much...time for me to make a donation to this cause!

**Reply by MAKOMO** [reply]
Thanks for your donation supporting our work on Artisan. I will try to find the cause of this password thing. I can only guess that there is some string encoding problem along the way. Not sure where it breaks yet...
